BedrockServer.cpp

```diff
@@ -418,6 +459,13 @@ void BedrockServer::sync(SData& args,
             SINFO("Sync thread dequeued command " << command.request.methodLine << ". Sync thread has "
                   << syncNodeQueuedCommands.size() << " queued commands.");
 
+            if (command.timeout() < STimeNow()) {
+                SINFO("Command '" << command.request.methodLine << "' timed out in sync thread queue, sending back to main queue.");
+                server._commandQueue.push(move(command));
```

Comment:
The main queue obeys the same rule that we added to the other three queues changed in this PR: Items that are past their `timeout` get returned first, regardless of where they are in the queue.

The actual place we throw `555 Timeout` here is when we try to peek the command again, as soon as a worker pulls it off the main queue (which will be soon, since it's timed out and cuts to the front of the line).